Common/src/main/java/com/blamejared/crafttweaker/api/zencode/impl/loader/ScriptRun.java

```diff
@@ -101,127 +89,81 @@ public void run() throws Exception {
         
         initializeBep();
-        registerModules();
+        configureRun();
         readAndExecuteScripts();
-        
     }
     
     private void initializeBep() {
         
         this.bep = new IgnorePrefixCasingBracketParser();
-        Services.EVENT.fireRegisterBEPEvent(bep);
-        
-        final List<Class<? extends IRecipeManager>> recipeManagers = CraftTweakerRegistry.getRecipeManagers();
-        bep.register("recipetype", new RecipeTypeBracketHandler(recipeManagers));
-        bep.register("constant", new EnumConstantBracketHandler());
-        
+        Services.EVENT.fireCustomRegisterBepEvent(this.scriptLoadingOptions.getLoaderName(), this.bep::register);
+    }
+    
+    private void configureRun() throws CompileException {
         
-        final TagManagerBracketHandler tagManagerBEP = new TagManagerBracketHandler(CrTTagRegistryData.INSTANCE);
-        bep.register("tagManager", tagManagerBEP);
-        bep.register("tag", new TagBracketHandler(tagManagerBEP));
+        this.scriptLoadingOptions.configureRun(this.bep, this.scriptingEngine);
     }
     
     private void readAndExecuteScripts() throws ParseException {
         
-        SemanticModule scripts = scriptingEngine.createScriptedModule("scripts", sourceFiles, bep, FunctionParameter.NONE);
+        final SemanticModule scripts = this.scriptingEngine.createScriptedModule("scripts", this.sourceFiles, this.bep, FunctionParameter.NONE);
         
         if(!scripts.isValid()) {
+            
             CraftTweakerAPI.LOGGER.error("Scripts are invalid!");
             CraftTweakerCommon.LOG.info("Scripts are invalid!");
             return;
         }
         
         //  toggle this to format scripts. Set by /ct format
-        if(scriptLoadingOptions.isFormat()) {
+        if(this.scriptLoadingOptions.isFormat()) {
+            
             writeFormattedFiles(scripts);
         }
         
-        if(scriptLoadingOptions.isExecute()) {
+        if(this.scriptLoadingOptions.isExecute()) {
+            
             //Now that we execute, we increment the runCount and therefore it's no longer a first run
             final LoaderActions loaderActions = getLoaderActions();
             CraftTweakerAPI.LOGGER.debug("This is loader '{}' run #{}", scriptLoadingOptions.getLoaderName(), loaderActions
                     .getRunCount() + 1);
             
-            scriptingEngine.registerCompiled(scripts);
-            scriptingEngine.run(Collections.emptyMap(), CraftTweakerCommon.class.getClassLoader());
+            this.scriptingEngine.registerCompiled(scripts);
+            this.scriptingEngine.run(Collections.emptyMap(), CraftTweakerCommon.class.getClassLoader());
             loaderActions.incrementRunCount();
             
         } else if(CraftTweakerAPI.DEBUG_MODE) {
-            scriptingEngine.createRunUnit().dump(new File("classes"));
+            this.scriptingEngine.createRunUnit().dump(Paths.get("classes").toFile());
```

Comment:
What was the difference between new Files directly and Paths.get again?
I remember falling back to `Paths.get` at some times but what was the issue again?